### ISSUE TITLE
`prefer-async-await` rule

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#123]
I see a lot of people doing:

```js
test(t => {
  return fetch().then(data => {
    t.is(data, 'foo');
  });
});
```

When this would be much succinct:

```js
test(async t => {
  t.is(await fetch(), 'foo');
});
```

Would be useful to have a rule to enforce using async/await.

**Comment by jfmengels** [comment]
How would we want to detect that? See if there's a `.then()` call somewhere in a normal function and there is a return statement? Don't know if it's that simple.

**Comment by sindresorhus** [comment]
I was thinking checking whether there's a `.then()` that is `return`'ed.

**Comment by jfmengels** [comment]
That could work well, only if we attempt to check the value of a variable if it's returned. This should be reported for instance
```js
test(t => {
  var foo = fetch().then(data => {
    t.is(data, 'foo');
  });
  return foo;
});
```

I'm starting to use `context.getScope()` in other projects. It's terribly undocumented, but even without code path analysis (though that'd be better probably), you can try and determine the value of a Node. When i have more time, I'll try to add it to this plugin. (I actually started working on it to make it work with macros, but then my computer crashed and I lost what I did ^^')

**Comment by sindresorhus** [comment]
> (I actually started working on it to make it work with macros, but then my computer crashed and I lost what I did ^^')

Oh man. That sucks. Backup though? 😜

**Comment by jfmengels** [comment]
> Backup though?

I wish ^^'
Apparently I don't have the habit of pushing my local work-in-progress branches to a repo, so... All in all, I only lost work on that macro thing for this repo, and some work in eslint-plugin-lodash-fp (a lot more unfortunately), don't think I really lost anything else except for the configuration for my computer. This should serve as a lesson I guess ^^' (and I should create a dotfiles project, soon)

**Comment by sindresorhus** [comment]
I don't usually push local work either, but I do have a close to real-time backup of my system using [BackBlaze](https://www.backblaze.com/). It has saved me many times.

**Comment by jfmengels** [comment]
> "Backup your Mac or PC just $5/month"

Linux :(
I may take a look at similar solutions soon though. Can be pretty useful :+1:
